ABIPacker interface + testing (#456)
ava-labs/teleporter: abi-bindings/README.md

## ABI Bindings

This directory contains ABI bindings for the Solidity contracts in the `contracts/src/CrossChainApplications` and `contracts/src/Teleporter` directories. The files with the same name as the Solidity source files are automatically generated by the `scripts/abi_bindings.sh` script. Other files in this directory (such as the packing utilities) are manually created and maintained.

## Packing

> `packing.go` files in individual subfolders define utilities for ABI packing instances of structs auto-generated by `abigen` as well as method calls. For structs we implement `ABIPacker` interface defined in `./packer/packer.go` and add references to the `packer_test.go` file to ensure that our tests are exhaustive and don't fail silently if additional fields are added to the structs in the future on the Solidty side.

Reviewer: generally have tried to avoid "we" and "our"

Author: Thanks. Made the update.

> This directory contains ABI bindings for the Solidity contracts in the `contracts/src/CrossChainApplications` and `contracts/src/Teleporter` directories. The files with the same name as the Solidity source files are automatically generated by the `scripts/abi_bindings.sh` script. Other files in this directory (such as the packing utilities) are manually created and maintained.
> 
> ## Packing
> 
> `packing.go` files in individual subfolders define utilities for ABI packing instances of structs auto-generated by `abigen` as well as method calls. For structs, `ABIPacker` interface defined in `./packer/packer.go` needs to be implemented and mapping to it's instance added to the `packer_test.go` file to ensure that the tests are exhaustive and don't fail silently if additional fields are added to the structs in the future on the Solidty side.

Reviewer 2: ```suggestion
This directory contains ABI bindings for the Solidity contracts in the `contracts/src/` directory. The files with the same name as the Solidity source files are automatically generated by the `scripts/abi_bindings.sh` script. 

The `packing.go` files in individual subfolders define utilities for ABI packing instances of structs auto-generated by `abigen` as well as method calls. For structs, the `ABIPacker` interface defined in `./packer/packer.go` needs to be implemented and mapped to its instance added to the `packer_test.go` file to ensure that the tests are exhaustive and don't fail silently if additional fields are added to the structs in the future on the Solidity side.
```
